library/lua/repeat-util.lua

```diff
@@ -17,7 +17,9 @@ function cancel(name)
  if not repeating[name] then
   return false
  end
- dfhack.timeout_active(repeating[name],nil)
+ if repeating[name] ~= -1 then
+  dfhack.timeout_active(repeating[name],nil)
```

Comment:
Yes, setting the timeout ID to nil prevents it from re-setting the timeout. The if on line 20 is just there to avoid an error message if `timeout_active(-1,nil)` ever starts responding with error messages.